### DISCUSSION TITLE
need user ID or Zep memory session ID in custom tools

### DISCUSSION THREAD

**Discussion by vijay-dhaked-rampwin** [FlowiseAI/Flowise#1058] [Q&A]
Hi,
I am building a custom chatflow, using Custom Tools to get name of user. I want to store that name in zep memory to tagged with that user id of that particular user.

Here is my chat flow:
![image](https://github.com/FlowiseAI/Flowise/assets/79076837/0350e25d-cefe-44f0-9744-a45e1bac378c)

I have "extract_name" custom tool, but in that i need zep session id.

so any one can pls guide me how can i get that, I've tried using curl as well.

Thanks

**Comment by mzmarkib** [comment]
I have the same problem, I cannot use the sessionId or the the userid in the inputs or the code, I want to be able to send the sessionId to an endpoint. please help!!!!

**Reply by HenryHengZJ** [reply]
This [PR](https://github.com/FlowiseAI/Flowise/pull/1422) will achieve this, need a day or two to review and get merged

**Reply by HenryHengZJ** [reply]
available now from v1.4.8!
